### ISSUE TITLE
Inquiry about the evaluation results of downstream tasks

### ISSUE THREAD

**Issue by szrrr04** [Caiyun-AI/DCFormer#20]
Hello, I previously followed the README in JAX exactly to pre-train a model on the Pile, and after about 15,000 steps, the model’s loss stabilized around 2.6. Afterwards, I converted it to PyTorch and evaluated it on multiple downstream task sets using the lm-evaluation-harness library. However, I noticed that the scores seemed less than satisfactory. I’d like to ask: is this because evaluating on downstream task sets (like RACE or SciQ) with the lm-evaluation-harness library requires fine-tuning first? Or is it because I didn’t train it enough (I trained it on roughly a 10GB subset of the Pile)? If fine-tuning is needed, could you please explain how to fine-tune it under the PyTorch model framework? If it’s due to insufficient training, how large of a Pile subset should I train on?

**Comment by szrrr04** [comment]
I think I forgot to mention earlier that I trained on a 10G subset of Pile, so I guess the reason might be insufficient pretraining? Should I increase it to 100G?

**Comment by hilbertmeng** [comment]
To make a fair comparison of downstream performance, you need to train different models with the same model size and compute. As model size and trained tokens increase, the downstream accuracy will go up. Like Pythia models, DCFormer-2.8B was trained on the whole Pile dataset (~300B tokens) to reach the performance as reported in the paper. If you only pursue high accuracy, you can directly use large open source models (eg. Llama or Qwen).

**Comment by szrrr04** [comment]
Thank you for your reply! I'm not aiming for high accuracy; instead, I'm trying to improve your DCFormer-405m and compare the performance of downstream tasks before and after the modifications. However, our lab has limited computational power, so training on the entire Pile dataset might be quite challenging.

Based on what you said, if I keep the model size the same and use a dataset of the same size (e.g., a 100GB subset of Pile), I should still be able to observe the differences in downstream task performance before and after the improvements, right?

**Comment by hilbertmeng** [comment]
Yes, I think you can see the difference if there is an improvement. However, it is not compute-efficient to train a small model on a large dataset. As the Chinchilla Scaling Law suggested, both model size and the amount of training data should scale at equal rates. Typically, the amount of trained tokens should be roughly 20x the model size. For example, it's compute-optimal to train a 0.4B model on 8B tokens. Thus, if compute resource is limited, you can run a scaling law experiment (matching the 20x relation) to compare validation loss, or set a more compute-efficient configuration for downstream tasks to demonstrate the effectiveness of your modification.

**Comment by szrrr04** [comment]
Thank you for your reply! So, do you mean that a 100GB subset of Pile is too large for my 0.4B model? According to what you said, if I match it to 8B tokens, then a Pile subset of around about 30GB should be sufficient, right? Then I can use this matched data size to train both the original and improved models simultaneously and determine whether my model has improved by comparing their validation losses, right?By the way, I’m very sorry, but I don’t quite understand what you mean by "more compute-efficient configuration for downstream tasks to demonstrate the effectiveness of your modification." Does this mean that I can no longer use the lm-evaluation-harness library for evaluation? Or does it mean that I should still use lm-evaluation-harness but with a different configuration?

**Comment by hilbertmeng** [comment]
Sorry to make the confusion. It has nothing to do with the lm-evalaution-harness repo. I mean you can choose a more compute-efficient **training** configuration for downstream tasks. For example, you may get higher downstream accuracy to train a 0.8B model on 16B tokens than a 0.4B model on 32B tokens, while both consume almost the same compute. However, it is not common to evaluate downstream tasks for small models(not well-trained models), because the general capability of small models is weak and it is likely to bring more variance on the accuracy. Instead, validation loss is a more stable metric for small models. Overall, it is challenging to use limited compute to validate the architectural improvement.

**Comment by szrrr04** [comment]
Thank you very much! I understand what you mean. One thing that confuses me is that I previously pre-trained dcformer-405m and my improved 405m model under the same conditions on the Pile subset using my lab's server. I found that after 15,000 steps, the original dcformer-405m had a loss of around 2.6, while my improved version's loss dropped to around 0.1.

I know this might sound a bit unbelievable, but first, I can confirm that overfitting should not be the issue, because for both the original dcformer and my improved version, the validation loss on val_with_eos after 13,500 steps was roughly the same as the final converged training loss.

I understand that something unusual might have happened, but I’m not sure exactly why.o(╥﹏╥)o
